lib/DXIL/DxilUtilDbgInfoAndMisc.cpp

```diff
@@ -109,7 +109,7 @@ Value *MergeGEP(GEPOperator *SrcGEP, GEPOperator *GEP,
   // Wrap the new gep in an addrspacecast if needed.
   if (AsCast)
     newGEP = Builder.CreateAddrSpaceCast(
-        newGEP, PointerType::get(newGEP->getType()->getPointerElementType(),
+        newGEP, PointerType::get(GEP->getType()->getPointerElementType(),
```

Comment:
Are newGEP and GEP different types?  Which scenario does switching newGEP for GEP fix?

Reply:
That is needed to support the change of type in the addrspacecast. The `newGEP` has the type of the result after the addrspacecast so when we generate the addrspacecase we have to use the original type.

Reply:
When reading through this code, I found myself wishing there were comments with examples showing what various branches are handling.  It's not easy to follow and recognize where test coverage might be missing, or the bugs might be.

Reply:
I agree that more comments would be great here. I tried to leave it better than I found it.

Reply:
Given the discussion about how we shouldn't merge GEPs around an addrspacecast that changes the pointer element type, I presume `newGEP` and `GEP` types will always be equivalent if we prevent that situation, right?

Reply:
Yes, they should be the same in that case. The GEP is RAWU newGEP just below here.

Reply:
I'm going to leave the code as is in case we want to support it in the future.